compiler/q-implant/src/QImplant.cpp

```diff
@@ -32,6 +32,10 @@ using namespace q_implant;
 
 namespace
 {
+// TODO: Find opcodes which can be appended in conversion
+std::set<luci::CircleOpcode> appendable_operator_opcode{
+  luci::CircleOpcode::CONCATENATION, luci::CircleOpcode::RESHAPE, luci::CircleOpcode::SPLIT,
```

Comment:
For concatenation, qparam has to be propagated backward.

PTAL `propagate_concat_quantparam` function in `PropagateQParamBackwardPass.cpp`.

Reply:
```
luci::PropagateQParamForwardPass forward_pass;
forward_pass.run(g);
```

If I understand `PropagateQParamForwardPass` correctly, running `PropagateQParamForwardPass` will propagate quantization parameter when operations like Reshape, Transpose.

```
bool copy_qparam(luci::CircleQuantParam *src, luci::CircleQuantParam *dst)
{
  assert(src->scale.size() == dst->scale.size());
  assert(src->zerop.size() == dst->zerop.size());
```

`copy_qparam` assume that source and destination node have same scale,zerop size, so we need to assign source's size into destination, right?

I'm not sure about it's correct or not `PropagateQParamForwardPass` maybe generally used in `quantized model` , so don't care about dtype. (Maybe the IR Graph is assumed as quantized correctly)

So, when I run the code by using `PropagateQParamForwardPass`, exception occured due to remaining float. (Quantization Parameter copied, but dtype not changed)

I not sure about changing dtype with using `PropagateQParamForwardPass` is possible or not.

<details>
<summary>Code</summary>
  std::set<luci::CircleNode *> visit;
  std::deque<luci::CircleNode *> que;
  for (auto input_node : loco::input_nodes(g))
  {
    auto node = loco::must_cast<luci::CircleNode *>(input_node);
    que.emplace_back(node);
  }

  while (!que.empty())
  {
    auto node = que.front();
    que.pop_front();

    if (node->opcode() == luci::CircleOpcode::CIRCLEOUTPUT)
    {
      continue;
    }

    auto node_qparam = node->quantparam();
    assert(node_qparam);

    for (auto child : loco::succs(node))
    {
      auto child_node = loco::must_cast<luci::CircleNode *>(child);
      if (visit.find(child_node) != visit.end())
        continue;
      visit.emplace(child_node);
      que.emplace_back(child_node);

      if (child_node->quantparam() == nullptr)
      {
        child_node->quantparam(std::make_unique<luci::CircleQuantParam>());
        auto child_qparam = child_node->quantparam();
        child_qparam->scale.assign(node_qparam->scale.size(), 0);
        child_qparam->zerop.assign(node_qparam->zerop.size(), 0);
      }

    }
  }

  luci::PropagateQParamForwardPass forward_pass;
  forward_pass.run(g);
</details>

Reply:
I see. So, we may need to do some preprocessing or modify `PropagateQParamForwardPass` to use it here. Thanks.

Reply:
Please remove concat from `appendable_operator_opcode`.